leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java

```diff
@@ -29,23 +36,116 @@
  */
 public class WriteRequest extends AbstractDownlinkRequest<WriteResponse> {
 
+    public enum Mode {
+        REPLACE, UPDATE
+    }
+
     private final LwM2mNode node;
     private final ContentFormat contentFormat;
+    private final Mode mode;
+
+    // ***************** write object ****************** //
+
+    public WriteRequest(final Mode mode, final ContentFormat contentFormat, final int objectId,
```

Comment:
Since there are so many constructors already, I'd rather have five more than sacrificing immutability.
Since Simon has introduced *static creator methods* inother places as well, we could instead think about turning these constructors into something like
```
public static new(final Mode mode, final int objectId ...) {
  return new WriteRequest(...);
}
...
public static new(....) {
  return ...
}
...
```
Another question is whether we actually need to know the content format at the time the request object is instantiated. Or can we postpone that decision to the time the object actually gets marshalled?